### PR TITLE
Add comparative benchmark of building simulation

### PR DESCRIPTION
**Pull request by BradLarson**
This introduces a comparative benchmark of gradient descent optimization of a simple thermal building simulation, implemented in differentiable Swift, PyTorch, and TensorFlow.

Instructions for how to perform the benchmarks are included, as well as current results.

### CODE REVIEW

**Comment by BradLarson** [comment]
cc @bjschoenfeld, @porterchild

**Comment by porterchild** [comment]
Thanks for moving these over Brad! 
Maybe these three comparisons should be in a subfolder within `Benchmarks`? I imagine we'll have other separate things in `Benchmarks`. like maybe ADBench...

**Comment by BradLarson** [comment]
@porterchild - Great point, I've moved this benchmark into a subdirectory to let us add others in the future.

**Review by JaapWijnen — COMMENTED** [review]
Looks good. I like the nesting of benchmarks. Ideally we'd add a proper swift benchmarking framework to make the numbers more robust. But we can do that in a next pass.

**Review by JaapWijnen — APPROVED** [review]
